linux.target

```diff
@@ -16,12 +16,14 @@ LIBOBJ = $(patsubst %.c,%.o,$(LIBOBJ_nodir))
 
 LIB_OBJECT = csmp_agent_lib.a
 
-LIBSRC += osal/linux/osal_common.c
-INC += osal/linux/
+LIBSRC += osal/linux/osal.c
+INC += \
+-I osal/linux/ \
```

Comment:
Do we really need that directory in the path? It is counterintuitive. Everything in that folder should have a local scope.

Reply:
I think yes, because I cannot use `#include ../osal.h` in osal_common.h since I have to specify the target dir: linux, freertos and later efr32. Relative path cannot be used by this way, because osal.h parent directory can be different based on different platforms

Reply:
If we have to do it it means that we are doing something wrong in osal_common.h. The public header should not be platform-dependent. The typedefs must be abstracted in osal_common.h and we should not need to include that platform dependent header.

Reply:
Indeed maybe we won't be able to do it here. It will not work cleanly with the thread id definition for instance. Still something bothers me. 

Consider that comment resolved, I'm completing my proposal about the new file architecture.